### PR TITLE
Improve resources validation

### LINKED ISSUE #60: Schema validation feature is broken
It's probably mismatching names, mostly (`gqlschemas` vs `schemas` etc)
Probably an artifact of #54

### PR DESCRIPTION
**Pull request by AleF83**
This PR fixes validation flow that is being invoked during creating new Agogos object.

Closes #60

### CODE REVIEW

File: services/registry/src/validation/index.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    upstream: validateUpstream,
+    upstreamclientcredentials: validateUpstreamClientCredentials,
+};
+
+
+export const validateNewObject = async (name: string, kind: string, source: string, spec: AgogosObjectConfig): Promise<void> => {
+    if (!validators.hasOwnProperty(kind)) {
+        throw new Error("Unknown GraphQL object kind");
+    }
+    await validators[kind].call(null, source, name, spec);
```
Comment (AvivRubys) [inline services/registry/src/validation/index.ts]: Why the `.call`?

File: services/registry/src/sync/sync-grpc-server.ts [hunk 2]
```diff
@@ -1,96 +1,97 @@
...
+                // TODO: extract to function
+                const upstreams: Upstream[] = Object.values(
+                    configuration.upstreams
+                ).map(ep => {
+                    const upstream = new Upstream();
+                    upstream.setHost(ep.host);
 
-          const upstreamAuth = new UpstreamAuthentication();
-          // FIXME: get from ep
-          upstreamAuth.setAuthType(ep.auth.authType);
```
Comment (AvivRubys) [inline services/registry/src/sync/sync-grpc-server.ts]: Types say `auth`, CRD says `authType`
Verify this is correct please :)

File: services/registry/src/validation/validators/upstreamValidator.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { AgogosObjectConfig, UpstreamConfig } from "../../sync/object-types";
+
+const validateUpstream = async (
+    source: string,
+    name: string,
+    spec: AgogosObjectConfig
+): Promise<void> => {
+    const upstreams = await gqlObjects$
+        .pipe(
+            map(x => x.upstreams || {}),
```
Comment (AvivRubys) [inline services/registry/src/validation/validators/upstreamValidator.ts]: Does this `map`->`take(1)` pattern used in all validators not block deployment when agogos is up but doesn't have any CRDs deployed?

File: services/registry/src/sync/sync-grpc-server.ts (hunk 2 above)
Comment (AleF83) [reply services/registry/src/sync/sync-grpc-server.ts]: It works so I think all things are correct

File: services/registry/src/validation/validators/upstreamValidator.ts (hunk 3 above)
Comment (AleF83) [reply services/registry/src/validation/validators/upstreamValidator.ts]: There is `startWith` [here](https://github.com/Soluto/agogos/blob/7c60e86149d9e84920a6431a2b313f90dfaf9492/services/registry/src/sync/sync-service.ts#L70).
So it shouldn't be empty ever

**Review by AvivRubys — APPROVED** [review]

**Pull request merged**